### PR TITLE
notification system

### PR DESCRIPTION
**Pull request by Valecant**
added the possibility to write
'default', 'success','error','warning','info'

is documented

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ import PolicyPage from './pages/policyPage'
 import { BrowserRouter, Routes, Route, NavLink } from 'react-router-dom'
 import jwt_decode from 'jwt-decode'
 import UserMenu from './components/shared/userMenu'
+import { SnackbarProvider } from "notistack";
```
Comment (chicco785) [inline src/App.js:39]: the coding style is different, and this should have triggered an error

File: src/components/policy/policyForm.js [hunk 2]
```diff
@@ -40,9 +40,13 @@ export default function PolicyForm({
   getServices,
   data
 }) {
+  const [msg, sendNotification] = useNotification();
```
Comment (chicco785) [inline src/components/policy/policyForm.js:43]: the coding style is different, and this should have triggered an error

File: src/components/shared/messages/alerts.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            enqueueSnackbar(conf.msg, {
+                variant: variant,
+                autoHideDuration: 5000,
+            });
+        }
+    },[conf]);
+    return [conf, setConf];
+};
+
+export default useNotification;
```
Comment (chicco785) [inline src/components/shared/messages/alerts.js:24]: line at the end is missing, plus the coding style is different, and this should have triggered an error

**Comment by chicco785** [comment]
@Valecant don't bother the style, we will change the linting rules to fix it

**Comment by chicco785** [comment]
@Valecant aren't supposed the errors to surface, for example, if the connection to the api fails? i stopped the api, the console reports the errors, but nothing happens in the ui

**Comment by chicco785** [comment]
@Valecant check commit 533ee753c0e56c4cde52e3e0d0210cb67e7ed7ae

i tried to:
- surface connectivity errors
- use info to inform on success (for the change of user preference)

my code is just an example, but should help you to apply the same principles to the whole app.

for sure there are other client (at least apollo in app.js and maybe an axiom too) that don't open a pop up in case of errors, please tackle that

File: src/components/shared/userMenu.js [hunk 4]
```diff
@@ -44,14 +44,14 @@ const CustomDialogTitle = styled(AppBar)({
...
 
-const [msg, sendNotification] = useNotification()
-console.log(msg)
 
 export default function UserMenu ({ language, userData, token }) {
   const { i18n } = useTranslation()
   const [anchorEl, setAnchorEl] = React.useState(null)
   const [settings, setOpenSettings] = React.useState(false)
-
+  const [msg, sendNotification] = useNotification()
```
Comment (chicco785) [inline src/components/shared/userMenu.js:52]: i moved them out, because when including them in the function, storybook fails

Comment (Valecant) [reply src/components/shared/userMenu.js:52]: storybook component needs to be mocked with the higher level component on the render return.
`
 <SnackbarProvider maxSnack={5}><Mycomponent></Mycomponent></SnackbarProvider>`

i'm fixing that on a new commit, just let me have the time to add a connection lost chip.

**Comment by Valecant** [comment]
If the API's or Graphql are offline, the app is going to show a banner, and try to reconnect, if the services are again back online the app will automatically reconnect.

**Comment by chicco785** [comment]
> If the API's or Graphql are offline, the app is going to show a banner, and try to reconnect, if the services are again back online the app will automatically reconnect.

graphql, as by meeting discussion, should not be a blocking api. not need to fix this in this pr, can be tackled in relation to #38 

in a software architecture you have essential dependencies, and dependencies that are not essential.
the ui should work fine with only:
1. keycloak
2. management-api

if graphql is down, we just show a message "preferences cannot be stored since configuration api is not reacheable"

**Review by chicco785 — APPROVED** [review]

**Pull request merged**